### DISCUSSION TITLE
Getting a pwm for JASPAR2024 - not working?

### DISCUSSION THREAD

**Discussion by johnblair7** [stuart-lab/signac#1646] [Q&A]
Hi all,

I know this isn't a signac issue per se but it affects how we might be using the tool. I recently installed the new JASPAR package (JASPAR2024) to get some of those sweet, new motifs, however now my code (which relies on TFBStools) to get the pwm from JASPAR doesn't work! It works with JASPAR2022 and JASPAR2020 so it must be something different with how the 2024 version is organized. I am really stupid so obviously I can't figure this out - has anyone else had this issue and how did you solve it?

`> pwm_set <- getMatrixSet(x = JASPAR2022, opts = list(species = 9606, all_versions = FALSE))
> pwm_set <- getMatrixSet(x = JASPAR2024, opts = list(species = 9606, all_versions = FALSE))
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function 'getMatrixSet' for signature '"function"'`
  
  Thanks for the help!!!

**Comment by timoast** [comment]
I also tried this a while ago and got the same error, I think the issue is that TFBSTools does not support JASPAR2024 yet. Hopefully they can release an update soon to add support.

**Comment by nzhang89** [comment]
You can follow this method to load JASPAR2024 database: [https://www.biostars.org/p/9588108/](https://www.biostars.org/p/9588108/). Hope this helps.

```
library(JASPAR2024) 
library(TFBSTools)
jaspar <- JASPAR2024()
sq24 <- RSQLite::dbConnect(RSQLite::SQLite(), db(jaspar))
motifs24 <- TFBSTools::getMatrixSet(sq24, list(species = "Homo sapiens", collection = "CORE"))
```
